python/cudf/cudf/_lib/copying.pyx

```diff
@@ -807,12 +809,13 @@ cdef class PackedColumns:
 
         p.c_obj = move(data_)
         p.column_names = header["column-names"]
+        p.column_dtypes = header["column-dtypes"]
         p.index_names = header["index-names"]
 
         return p
 
     @staticmethod
-    cdef PackedColumns c_from_py_table(Table input_table, keep_index=True):
+    cdef PackedColumns from_py_table(Table input_table, keep_index=True):
```

Comment:
Any reason for this (and `unpack`) below to be `cdef` methods as opposed to `def` methods? If we make them `def` methods, they can be called from Python directly, without the need for the trivial `pack` and `unpack` free functions below.

Reply:
I made the trivial functions in an attempt to mirror the C++ API for pack/unpack with static functions outside of the `cudf::packed_columns` namespace; the `cdefs` here are unnecessary though, thanks for catching that.

If mirroring that aspect of the C++ API isn't important, maybe I could also rename this function `pack` so that the Python API would look something like:

```python
from cudf._lib.copying import PackedColumns

packed = PackedColumns.pack(df)
new_df = packed.unpack()
```

Reply:
Hmm... yeah on second thought, I like the free functions better, in which case leaving these as `cdef` methods might be a bit more performant.